axum-macros/tests/from_request/pass/state_infer.rs

```diff
@@ -0,0 +1,18 @@
+use axum_macros::FromRequest;
+use axum::extract::State;
+
+#[derive(FromRequest)]
+struct Extractor {
+    inner_state: State<AppState>,
```

Comment:
You made the derive take the first `State<_>` field even if there are multiple, right? Would be good to also have a test for that so it doesn't accidentally regress.

Reply:
You mean something like

```rust
#[derive(FromRequest)]
struct Extractor {
    inner_state: State<AppState>,
    also_inner_state: State<OtherState>,
}
```

There it'll require you to specify the state because there are two different options.

I tried to make that work with generating bounds like `AppState: FromRef<S>` and `OtherState: FromRef<S>` but ran into issues with that that broke other things I think 🤔

Reply:
Maybe have a failing UI test for that case then? Or is there already one?

Reply:
Sure. I've added one.